### PR TITLE
Bump various GitHub Actions, handle new `jupyter_core`

### PR DESCRIPTION
**Pull request by jtpio**
## References

Perform a bit of maintenance on the GitHub Actions to fix some of the following warning:

image

## Code changes

- [x] Bump various actions
- [x] Add `dependabot.yml` to make this process more automated: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot#example-dependabotyml-file-for-github-actions
- [x] Handle new `jupyter_core`: https://github.com/jupyter/jupyter_core/commit/1118c8ce01800cb689d51f655f5ccef19516e283

## User-facing changes

None

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by jtpio** [comment]
Looks like the CI failures need proper investigation (also failing on `main`).

**Comment by jtpio** [comment]
`jupyter_core==4.11.2` seems to be cause of the failures. So probably related to this commit: https://github.com/jupyter/jupyter_core/commit/1118c8ce01800cb689d51f655f5ccef19516e283

**Comment by jtpio** [comment]
Most deprecation warnings should now be gone (still a couple of left, will have a look another time).

File: py/jupyterlite/src/jupyterlite/app.py [hunk 1]
```diff
@@ -80,6 +85,10 @@ class BaseLiteApp(JupyterApp, LiteBuildConfig, DescribedMixin):
 
     flags = lite_flags
 
+    @default("config_file_paths")
+    def _config_file_paths_default(self):
+        return [str(Path.cwd())] + jupyter_config_path()
```
Comment (bollwyvl) [inline py/jupyterlite/src/jupyterlite/app.py:90]: yep, let's get rid of cwd.

Comment (bollwyvl) [reply py/jupyterlite/src/jupyterlite/app.py:90]: it'll probably break some tests/binder, though.

Comment (jtpio) [reply py/jupyterlite/src/jupyterlite/app.py:90]: ok let's do the change in its own PR so it's more visible in the changelog.

I guess that means `--config jupyter_lite_config.json` should now be used explicitly.

Comment (bollwyvl) [reply py/jupyterlite/src/jupyterlite/app.py:90]: ugh. a good point. i guess we need to think about it a bit more: running `jupyter lite build` from a CLI is a different activity altogether than browsing a web page.

**Review by bollwyvl — APPROVED** [review]

**Comment by jtpio** [comment]
Following up with a new release since new deployments are likely affected by the latest `jupyter_core` release.

**Pull request merged**